account-based-state.md

```diff
@@ -1,4 +1,52 @@
+
+
+### State Confidentiality between Smart contracts
+
+The main goal of Obscuro is to protect user data. If smart contracts were wholly isolated from each other, it would be easy to define data access rules.
+
+For example, for an individual (non-contract) account, the main rule could be formalized as:
+*"Only the account owner, or someone empowered by the owner, is allowed to query the balance of an account."*
+
+Contract composition introduces significant complexity. DeFi enjoys massive success thanks to the ability of contracts to be combined in serendipitous ways not predicted by the contract creator, and Obscuro intends to replicate that.
+
+For example, consider a flash-loan where a user borrows 10 ETH from Aave, swaps them for 100 ABC coins on UniSwap, then swaps them for 500 DEF coins on SushiSwap, and then swaps them for 10.1 ETH on UniSwap, and then pays back the loan and makes a 0.1 ETH profit.
+
+In this case, the flash-loan contract needs to query balances or find out if it can loan a certain amount. It also needs to know the price for the ABC/BCD on a specific exchange. All this information that it needs access to perform its duties could, in theory, be emitted directly or indirectly as an event to the caller of the contract. Alternatively, the attacker could write the state on accounts she controlled.
+
+This area is still under active research. The first version will rely on application developers to check programmatically who can call different functions and what data they should receive.
+
+For subsequent versions, Obscuro will explore the following concepts:
+* Each contract can declaratively whitelist contracts that can access different functions.
+* Automatically propagate access. For example, if _Account.getBalance()_ can be invoked only by the owner, it means that any contract that invokes this has to originate from a message signed by the owner. This solution sounds appealing, but it needs more research to determine if this mechanism prevents useful use cases.
+
+###  Wallets and Transaction Submission
+User wallets created transactions encrypted with the shared _Obscuro public key_. Only valid TEEs in possession of the private key can decrypt, execute, and see the resulting state. Still, end-users who submitted a transaction must be able to receive the result and query the balance.
```

Comment:
Should this read: "User wallets create transactions..."

Reply:
yes